File: stock/src/main/java/com/trifail/stock/resource/StockResource.java
package com.trifail.stock.resource;

import com.trifail.protocol.core.ErrorCode;
import com.trifail.stock.common.StockErrorCode;
import com.trifail.stock.service.IGoodsService;
import model.V1GoodWrapper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;
import service.ApiStockService;


@RestController
public class StockResource implements ApiStockService {

    private final IGoodsService goodsService;

    @Autowired
    public StockResource(IGoodsService goodsService) {
        this.goodsService = goodsService;
    }


    @Override
    public ErrorCode checkGoodsWithStock(@RequestBody V1GoodWrapper goodsInfo) {
        System.out.println("chenggong");
        return StockErrorCode.GOODS_NOT_EXISTS;
    }

    @Override
    public boolean deliverGoods(@RequestBody V1GoodWrapper goodsInfo) {
        return false;
    }
}

File: stock/src/main/kotlin/com/trifail/store/resource/StockResource.kt
package com.trifail.store.resource

import com.trifail.store.databean.V1GoodsInfo
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/v1")
class StockResource {

    @Autowired
//    private lateinit var goodService: IGoodsService

//    @PostMapping("/validation")
//    fun checkGoodsWithStock(@RequestBody goods: List<V1GoodsInfo>):List<V1GoodsInfo>{
//    }

    /**
     * 出库
     */
    @PostMapping("/deliver")
    fun deliverGoods(): Any {
        return Any();
    }

}